File: freak-fights-fan/shared/features/fights/responses/FighterProfileDto.ts
import { FightResult } from "../helpers/fightResult";
import { FightDto } from "./FightDto";

export interface FighterProfileDto {
    profileFights: Array<ProfileFightDto>;
    stats: FighterFightsStats;
}

export interface ProfileFightDto {
    fight: FightDto;
    fightResult: FightResult;
}

export interface FighterFightsStats {
    upcoming: number;
    win: number;
    loss: number;
    draw: number;
    noContest: number;
}

File: FreakFightsFan.Shared/Features/Fights/Responses/FighterProfileDto.cs
﻿using FreakFightsFan.Shared.Features.Fights.Helpers;

namespace FreakFightsFan.Shared.Features.Fights.Responses;

public class FighterProfileDto
{
    public List<ProfileFightDto> ProfileFights { get; set; }
    public FighterFightsStats Stats { get; set; }
}

public class ProfileFightDto
{
    public FightDto Fight { get; set; }
    public FightResult FightResult { get; set; }
}

public class FighterFightsStats
{
    public int Upcoming { get; set; }
    public int Win { get; init; }
    public int Loss { get; init; }
    public int Draw { get; init; }
    public int NoContest { get; init; }
}